app/models/name/parse.rb

```diff
@@ -590,7 +590,11 @@ def clean_incoming_string(str)
       gsub(/‘|’/, "'").
       delete("\u2028"). # Unicode RLE that we see occasionally as line separator
       gsub(/\s+/, " ").
-      strip_squeeze
+      strip_squeeze.
+      # remove trailing spacing characters including non-breaking spaces
+      sub(/[\s\u00A0\u202F]+\z/, "").
+      # and the same for leading
+      sub(/\A[\s\u00A0\u202F]+/, "")
```

Comment:
@nwilson-eol: I think we should do the same thing for the beginning of the string.
Does that makes sense?
If so, should I add a test for that?
Also, what about the middle of the incoming string. (I left that as is because I don't think a nonbreaking space in the middle will break anything.)